Content/Items/Misc/Weapons.Earthduster.cs

```diff
@@ -0,0 +1,530 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Content.Items.Misc.SoilgunFiles;
+using StarlightRiver.Core;
+using StarlightRiver.Core.Systems;
+using StarlightRiver.Helpers;
+using System;
+using System.Collections.Generic;
+using Terraria;
+using Terraria.Audio;
+using Terraria.DataStructures;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace StarlightRiver.Content.Items.Misc
+{
+    public class Earthduster : MultiAmmoWeapon
+    {
+        public override List<AmmoStruct> ValidAmmos => new List<AmmoStruct>()
+        {
+            new AmmoStruct(ItemID.SandBlock, ModContent.ProjectileType<SoilgunSandSoil>(), 2),
+            new AmmoStruct(ItemID.EbonsandBlock, ModContent.ProjectileType<SoilgunEbonsandSoil>(), 4),
+            new AmmoStruct(ItemID.PearlsandBlock, ModContent.ProjectileType<SoilgunPearlsandSoil>(), 15),
+            new AmmoStruct(ItemID.CrimsandBlock, ModContent.ProjectileType<SoilgunCrimsandSoil>(), 4),
+            new AmmoStruct(ItemID.DirtBlock, ModContent.ProjectileType<SoilgunDirtSoil>(), 2),
+            new AmmoStruct(ItemID.SiltBlock, ModContent.ProjectileType<SoilgunSiltSoil>(), 3),
+            new AmmoStruct(ItemID.SlushBlock, ModContent.ProjectileType<SoilgunSlushSoil>(), 3),
+            new AmmoStruct(Mod.Find<ModItem>("VitricSandItem").Type, ModContent.ProjectileType<SoilgunVitricSandSoil>(), 8),
+            new AmmoStruct(ItemID.MudBlock, ModContent.ProjectileType<SoilgunMudSoil>(), 3),
+        };
+
+        public override string Texture => AssetDirectory.MiscItem + Name;
+
+        public override void SetStaticDefaults()
+        {
+            Tooltip.SetDefault("Hold <left> to fire a rapid stream of soil\nCan use many different types of soils\n33% chance to not consume ammo");
+        }
+
+        public override void SafeSetDefaults()
+        {
+            Item.DamageType = DamageClass.Ranged;
+            Item.damage = 25;
+            Item.width = 70;
+            Item.height = 40;
+            Item.useAnimation = Item.useTime = 5;
+            Item.shoot = ProjectileID.PurificationPowder;
+            Item.useStyle = ItemUseStyleID.Shoot;
+            Item.rare = ItemRarityID.Orange;
+            Item.shootSpeed = 21f;
+            Item.noUseGraphic = true;
+            Item.noMelee = true;
+            Item.channel = true;
+            Item.knockBack = 3f;
```

Comment:
this should also have Item.autoReuse = true; so you can keep holding click and have it keep firing after reloading